.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  release:
+    types: [created]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        scala_version: &scala_versions
+          - 2.12.12
+          - 2.13.4
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with: &cache_options
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ matrix.scala_version }}
+          restore-keys:
+            - ${{ runner.os }}-sbt-${{ hashFles('**/*.sbt') }}-
+      - name: Build
+        run: sbt ++${{ matrix.scala_version }} compile packageBin packageDoc packageSrc
+      - name: Test
+        run: sbt ++${{ matrix.scala_version }} test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: target-${{ matrix.scala_version }}
+          path: '*/target/'
+          retention-days: 1
+  publish:
+    if: github.event_name == "push" || github.event_name == "release"
+    needs: build-and-test
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        scala_version: *scala_versions
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with: *cache_options
+      - uses: actions/download-artifact@v2
+        with:
+          name: target-${{ matrix.scala_version }}
+      - name: Publish to Sonatype
+        run: bash publish.sh
+        env:
+          SCALA_VERSION: ${{ matrix.scala_version }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+  release-upload:
+    if: github.event_name == "release"
```

Comment:
iiuc, this job automatically creates a release for this tag. But this only runs on the `release - created` event; did you mean to have it run on tag push?

Reply:
Oh... this job edits the existing release. I see. Should we have the job run on tag push and create the release too? It seems that was what Travis was doing.